### ISSUE TITLE
ATTENTION: Service Termination of Your meross Smart Device due to Security Concerns

### ISSUE THREAD

**Issue by WahoYate** [albertogeniola/meross-homeassistant#251]
Received today from Meros : 

"Dear Customer,

This is meross cloud security team contacting you.
First of all, thanks for purchasing meross and we hope that you enjoy life with our smart products.
Recently our cloud security system has noticed that your devices are communicating with the could server at an extremely high frequency. We consider these behaviors abnormal and are concerned about the security of your devices.
Due to the security concerns and the intent to avoid any further potential damage, we will terminate the cloud services of these devices for now.
If you are sure that your devices are safe, NOT hacked, and are performing as expected. **Please first lower your periodic request rate to no more than one message every ten seconds** and then send an email to <support@meross.com>. We will resume cloud service for you within 24 hours. Please note that any abnormal high cloud request frequency will cause cloud service termination again.
Should you have any information you would like to provide us with or would like to offer us any clarification, please contact us within 24 hours.

Thanks!
meross Cloud Security Team  "

How to lower my periodic request rate to no more than one message every ten seconds ?
why it's not default configuration ?
Tanks for your advices.

**Comment by zoccaz** [comment]
I have the same issue since few days, received the same email, and since my HA meross intégration is down 😒

**Comment by eximo84** [comment]
So i have been having issues with the devices becoming unavailable. I decided to update the integration which hasn't helped. I've then decided to update the firmware and now the integration won't connect to meross cloud. 

I haven't received any email but now I can't use my devices via HA

**Comment by zoccaz** [comment]
someone has a solution ?

**Comment by purcell-lab** [comment]
> someone has a solution ?
>
I migrated to the Meross LAN HA integration and now have local control and
no issues with what they do upstream.

**Comment by rpitera** [comment]
I liked the Meross outdoor switch but when I found out I could integrate it into HA, I was super happy and I bought another one right away.  The first one was working great and when the second arrived today, I immediately set it up in the app.  I saw a firmware update for the first one was available and stupidly applied it.  Now that switch no longer works with HA.  The new one is working fine and so far haven't gotten any emails like the OP but fearfully expecting one.

TL;DR - DO NOT apply any firmware updates as they seem to be specifically to block HA.  :(

Probably going to have to look into @purcell-lab's suggestion... too bad because I was really happy with the way this integration worked and it was the MAIN reason I made another Meross purchase.

**Comment by iBeech** [comment]
I’ve put a fix in that was causing the integration to hit the servers too often. It may fix this issue. PR is out

**Comment by rpitera** [comment]
Well, that's great and thanks!  But to be honest, the old switch started working with HA again so I wonder if they changed a little with their code or policies on the Meross end?  Either way, I'm happy.

**Comment by albertogeniola** [comment]
As you probably know, Meross applies API rate limits without any notice. I've released a new version (v1.1.0 ) of the component which allows the user to configure API rate limiting from the integration option menu. There is no "magic" API limit logic that works for everyone: it depends on the automation scripts and on the number of Meross devices installed. 

You can rely on the Meross LAN integration, or wait for the Meross Local-Lan addon which is currently work in progress.
